android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -409,8 +420,20 @@ constructor(
       )
       return
     }
-
     defaultRepository.addOrUpdate(resource = questionnaireResponse)
+    if (actionParamUpdatableId != null) {
+      val resource =
+        defaultRepository.loadResource(
+          actionParamUpdatableId!!.value.extractLogicalIdUuid(),
+          actionParamUpdatableId!!
+            .value
+            .substringBefore("/")
+            .resourceClassType()
+            .newInstance()
+            .resourceType
+        )
```

Comment:
haven't we already verified it's not null? might also use a temp var

```suggestion
      val resource = actionParamUpdatableId.value.let {
        defaultRepository.loadResource(
          it.extractLogicalIdUuid(),
          it
            .substringBefore("/")
            .resourceClassType()
            .newInstance()
            .resourceType
        )
      }
```